### PR TITLE
Fix HTTPS client host no domain issue

### LINKED ISSUE #2422: gscan/gcylc: launching gcylc via gscan incorrectly shows suite as stopped
Reported by user launching launching gcylc by double clicking on a suite in gscan.

(Since 7.5)

Reproducible using same steps:
1. Start a test suite running
2. Launch gscan
3. Launch gcylc by double-clicking the suite in gscan
4. In gcylc, the suite shows as "Stopped", even when the suite is still running.

Launching gcylc from the command line does not produce the issue (only via gscan)

Note that the domain name is omitted when launched via gscan (top image), but not when launched via the command line (bottom image). 

launch-differ

### PR DESCRIPTION
**Pull request by matthewrmshin**
A CLI specified host name with no domain part was causing the HTTPS
client to fail because it is unable to match the host name in the SSL
certificate (in which the host name has the domain name).

Rename `cylc.suite_host` to `cylc.hostuserutil` (as the module is now a generic utility for working with hosts and users).

Should fix #2422.

### CODE REVIEW

File: lib/cylc/hostuserutil.py [hunk 1]
```diff
@@ -112,43 +111,43 @@ def get_host_ip_by_name(target):
...
             elif method == 'hardwired' and hardwired:
-                self._host_name_pref = hardwired
+                self._host = hardwired
             else:  # if method == 'name':
-                self._host_name_pref = self.get_hostname()
-        return self._host_name_pref
+                self._host = self._get_host_info()[0]
+        return self._host
+
+    def get_fqdn_by_host(self, target):
```
[the remark this answers is not in the document]
Comment (matthewrmshin) [reply lib/cylc/hostuserutil.py]: *Fully qualified domain name* is probably too long? And *fqdn* matches `socket.getfqdn`, which is *hopefully* what this method is returning.

**Comment by oliver-sanders** [comment]
The hostname as it appears in the title of the `gcylc` window is different when `gcylc` is launched from the command line to when it is launched from `gscan`.

This fixes #2422 for me but I have found a new similar bug which can be repeated with these steps:

1) Run a suite on another host.
2) Open `gcylc <suite>` from the command line.
3) Open another `gcylc` window from `gscan`.
4) Stop the suite.
5) Restart the suite from the first `gcylc` window.

**Comment by dvalters** [comment]
> The hostname as it appears in the title of the gcylc window is different when gcylc is launched from the command line to when it is launched from gscan.

Yes I get this too - it is missing the domain name qualifier when launched via gscan - it only prints the cylc host.

**Comment by matthewrmshin** [comment]
(Should have gone for my original implementation, which would have eliminated all of these issues.)

File: lib/cylc/hostuserutil.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvalters) [reply lib/cylc/hostuserutil.py]: It could just go in the docstring then

**Comment by matthewrmshin** [comment]
(Should now work?)

**Comment by dvalters** [comment]
Yep, the gcylc title outputs are matching now. @oliver-sanders bug looks to be fixed too.

**Comment by matthewrmshin** [comment]
(Conflicts resolved.)

**Comment by matthewrmshin** [comment]
@dvalters please do a final sanity check.

**Comment by matthewrmshin** [comment]
(All looking healthy again.)

**Comment by matthewrmshin** [comment]
(1 final tweak. Waiting for test.)

**Comment by matthewrmshin** [comment]
(Merge as all green with double review approval.)

**Pull request merged**